### PR TITLE
Ox trust/issues/#785 

### PR DESCRIPTION
**Pull request by earezki**

### CODE REVIEW

**Comment by syntrydy** [comment]
What is the purpose of this commit https://github.com/GluuFederation/oxTrust/pull/1022/commits/e55702732443483ef13485146b22efdaeebf97bf

Please update your branch with the latest changes in **oxtrust_api** branch

File: client/conf/configuration.properties [hunk 1]
```diff
@@ -2,6 +2,6 @@
 # oxTrus API integration test application configuration file
```
Comment (yurem) [inline client/conf/configuration.properties]: We should use `profiles` here to allow change test data without updating file.
In oxTrust/server we uses this approach. You can use it as reference

File: server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationExtractor.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.gluu.oxtrust.service.config;
+
+import org.gluu.persist.PersistenceEntryManager;
+import org.xdi.config.oxtrust.LdapOxAuthConfiguration;
+
+public class OxAuthConfigurationExtractor {
+    private final PersistenceEntryManager persistenceEntryManager;
+
+    public OxAuthConfigurationExtractor(PersistenceEntryManager persistenceEntryManager) {
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationExtractor.java]: We should not pass and store reference to PersistenceEntryManager. CDI manages it lifecyсle. We need to use `@Inject` to get reference

File: server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationDAO.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private final OxAuthConfigDynamicExtractor oxAuthConfigDynamicExtractor;
+    private final ConfigurationFactory configurationFactory;
+    private final PersistenceEntryManager persistenceEntryManager;
+
+    public OxAuthConfigurationDAO() {
+        // required for proxying.
+        this(null, null, null, null, null);
+    }
+
+    public OxAuthConfigurationDAO(OxAuthConfigurationExtractor oxAuthConfigurationExtractor,
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationDAO.java]: Wrong method of service creation. We should use `@Inject`

File: server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationDAO.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package org.gluu.oxtrust.service.config;
+
+import org.gluu.oxtrust.api.configuration.OxAuthConfig;
+import org.gluu.oxtrust.config.ConfigurationFactory;
+import org.gluu.persist.PersistenceEntryManager;
+import org.xdi.config.oxtrust.LdapOxAuthConfiguration;
+
+import java.io.IOException;
+
+public class OxAuthConfigurationDAO {
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigurationDAO.java]: Let's continue to use old naming convention in this project. Use `Serivce` for reusable methods and `Acrtion` for GUI classes

File: server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigDynamicExtractor.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.io.IOException;
+
+public class OxAuthConfigDynamicExtractor {
+
+    private final ConfigurationFactory configurationFactory;
+    private final OxAuthConfigurationExtractor oxAuthConfigurationExtractor;
+    private final OxAuthConfigObjectMapper oxAuthConfigObjectMapper;
+
+    public OxAuthConfigDynamicExtractor(ConfigurationFactory configurationFactory,
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/service/config/OxAuthConfigDynamicExtractor.java]: Let's use CDI here

File: server/src/main/java/org/gluu/oxtrust/ldap/service/AppInitializer.java [hunk 6]
```diff
@@ -253,12 +258,26 @@ public PersistenceEntryManager createLdapEntryManager() {
...
         Properties decryptedConnectionProperties = securityService.decryptProperties(connectionProperties);
 
-        PersistenceEntryManager ldapEntryManager = this.ldapEntryManagerFactory.createEntryManager(decryptedConnectionProperties); 
+        PersistenceEntryManager ldapEntryManager = this.ldapEntryManagerFactory.createEntryManager(decryptedConnectionProperties);
         log.info("Created {}: {}", new Object[] { LDAP_ENTRY_MANAGER_NAME, ldapEntryManager.getOperationService() });
 
         return ldapEntryManager;
     }
 
+    @Produces
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/ldap/service/AppInitializer.java]: I think we can avoid this `@Produces`. Let's review OxAuthConfigurationDAO  according comments above

File: client/src/test/java/org/gluu/oxtrust/api/configuration/OxAuthConfigTest.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class OxAuthConfigTest {
+
+    @Test
+    public void url_validation_happy_path() {
+        String opPolicyUri = "http://ox.gluu.org/doku.php?id=oxauth:policy";
```
Comment (yurem) [inline client/src/test/java/org/gluu/oxtrust/api/configuration/OxAuthConfigTest.java]: we should move URL to test profile

File: client/src/main/java/org/gluu/oxtrust/api/test/TestMain.java [hunk 8]
```diff
@@ -39,7 +39,7 @@
      * @throws IOException
      */
     public void init() throws IOException {
-        final String confFile = "conf/configuration.properties";
+        final String confFile = "client/conf/configuration.properties";
```
Comment (yurem) [inline client/src/main/java/org/gluu/oxtrust/api/test/TestMain.java]: Let's take a look on how we work with profiles in `Server`

File: client/src/test/java/org/gluu/oxtrust/api/configuration/OxAuthConfigTest.java (hunk 7 above)
Comment (earezki) [reply client/src/test/java/org/gluu/oxtrust/api/configuration/OxAuthConfigTest.java]: Done for all the other remarks as requested.
For the tests, this is only for local usage, Thomas is already working on an automating testing app.

**Pull request merged**